### DISCUSSION TITLE
feat: Add Go Client documentation website

### DISCUSSION THREAD

**Discussion by ittuann** [apache/streampipes#2973] [Ideas]
I noticed that the Go client currently under development does not yet have accompanying documentation or a website. 

I therefore propose to use the Go-developed [Hugo](https://gohugo.io/) framework, combined with the [Docsy](https://www.docsy.dev/) theme to create a documentation site. Hugo is an efficient static site generator, and paired with Docsy, which is designed for technical documentation, it is well-suited for our project documentation needs.

For this purpose, I have quickly set up a demo site. If this proposal is approved, I would be willing to commit my current work to the project.

![image](https://github.com/apache/streampipes/assets/80267918/ce9ee7a8-684a-407e-a70f-91fe5df124ee)

**Comment by bossenti** [comment]
I really appreciate a proper documentation for the Go client, @ittuann 

Do you want to make the documentation part of our website or where should it be?

In general, I would like the style to be a bit more in line with our StreamPipes color scheme (primary and accent color, etc.)

**Reply by ittuann** [reply]
> I really appreciate a proper documentation for the Go client, @ittuann
> 
> Do you want to make the documentation part of our website or where should it be?
> 
> In general, I would like the style to be a bit more in line with our StreamPipes color scheme (primary and accent color, etc.)

Based on the format of the Python Client docs website, I modeled the Go Client docs website! This is exactly why I chose the Hugo framework developed in Go.
Therefore, I want to integrate the documentation to be part of the main website, accessible from the main site's dropdown menu, just like the Python Client docs website.

For the color scheme, I configured it as `#39B54A` and `#1B1464` in turn with reference to the Python client.

![image](https://github.com/apache/streampipes/assets/80267918/148f67c2-6fa4-464f-b54c-281b263f8a46)

**Comment by bossenti** [comment]
> Therefore, I want to integrate the documentation to be part of the main website, accessible from the main site's dropdown menu, just like the Python Client docs website.

Okay, great 👍🏼 

Yep, this screenshot looks much better!

So approved from my side 🙂 
How about the others?

**Reply by tenthe** [reply]
Also +1 from my side. @ittuann thanks for the effort.

**Comment by dominikriemer** [comment]
Hi @ittuann great that you are working on the documentation!
Appreciate your effort and I also support this proposal.
